include/server/SC_WorldOptions.h

```diff
@@ -49,6 +49,7 @@ struct WorldOptions {
 
     bool mRealTime = true;
     bool mMemoryLocking = false;
+    float mSafetyClipThreshold = 2.0;
```

Comment:
note -- look for convo w @Spacechild1 about ABI/API changes

Reply:
thanks @brianlheim for catching that! `WorldOptions` is part of the public libscsynth API and changing a public struct causes API/ABI breakage.

Note that the libscsynth API/ABI has already been broken in past. For example, https://github.com/supercollider/supercollider/commit/695aea5b932d5915eeca629803d9565016ff2e2d changed function signatures, and https://github.com/supercollider/supercollider/commit/22c9deac2e38f6446bf254c55157925852c96142 turned `WorldOptions` from a old-style C++ struct to a C++11 struct. The first commit causes a linker error if header and binary don't have the same version. The second commit has no consequences expect possible compilation errors for pre-C++11 projects. 

However, changing the layout of a public struct can cause runtime crashes, e.g. when the host uses a new libsynth binary with the old header. This means it is now required that the `SC_WorldOptions.h` header and the libscsynth binary have the same SC version (which means you can't update the libscsynth binary without recompiling the host app against the same SuperCollider SDK version, and vice versa!) This is just something to be aware of.

I see two solutions:
a) make it clear that header and binary must have the same version
b) add a simple versioning mechanism similar to the SC plugin API, so that `World_New` fails on a version mismatch.

Reply:
thanks @Spacechild1 !!

i don't know if many people use this part of the API at all. just thinking back, i have never seen or heard any concerns over the shared-libscsynth build breaking -- except for the Arch Linux distribution, but David Runge said he's not aware of anyone making use of it either. i've also never seen complaints about crashes or other breakage when those API changes did happen.

i wouldn't want to spend time on adding a feature that nobody will make use of, so for now i think we could just document in the header that the API may change across minor versions, so it should be compiled against headers from the same minor version used to link. that's at least a small stability guarantee. what do you think?

Reply:
> i don't know if many people use this part of the API at all

I'm sure there are a few people who use libscsynth, but they probably all use a binary + header with the same version. So in practice, ABI breakage is not such a big deal.

> i've also never seen complaints about crashes or other breakage when those API changes did happen.

The difference is that previous breakages would lead to a compiler resp. linker error, which are easy to spot/fix. Changing the struct can cause runtime crashes, which are harder to debug.

> so for now i think we could just document in the header that the API may change across minor versions, so it should be compiled against headers from the same minor version used to link. that's at least a small stability guarantee. what do you think?

That would be option a), and I think that's probably sufficient :-)

Reply:
@elgiano can you please add this note?

Reply:
Something like this?
```
// PLEASE NOTE:
// This API might change across minor versions.
// Always make sure, when using libscsynth, that the binary and this header come from the same minor version.
#